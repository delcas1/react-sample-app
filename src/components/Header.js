import React from 'react';
import { NavLink } from 'react-router-dom';

//Navlink component better 
const Header = () => (
  <header>      
       <div>
        <h1>Test-React-App</h1>
        <NavLink to="/"activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create CRUD</NavLink>
      {/*   <NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink> */}
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
      </div>
      
  </header>    
);
export default Header;
