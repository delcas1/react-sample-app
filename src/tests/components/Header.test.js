import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();  
});

//expect(wrapper.find('h1').text()).toBe('Main');
//expect(wrapper).toMatchSnapshot();
test('should find the correct title in the h1', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find('h1').text()).toBe('Test-React-App');  
});

