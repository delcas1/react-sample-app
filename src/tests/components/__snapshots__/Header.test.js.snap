// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <div>
    <h1>
      Test-React-App
    </h1>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Create CRUD
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/help"
    >
      Help
    </NavLink>
  </div>
</header>
`;
